torchbearer/callbacks/imaging.py

```diff
@@ -0,0 +1,328 @@
+            This will be applied to the image before it is sent to output.
+        num_images: The number of images to cache
+        nrow: See `torchvision.utils.make_grid <https://pytorch.org/docs/stable/torchvision/utils.html#torchvision.utils.make_grid>`_
+        padding: See `torchvision.utils.make_grid <https://pytorch.org/docs/stable/torchvision/utils.html#torchvision.utils.make_grid>`_
+        normalize: See `torchvision.utils.make_grid <https://pytorch.org/docs/stable/torchvision/utils.html#torchvision.utils.make_grid>`_
+        norm_range: See `torchvision.utils.make_grid <https://pytorch.org/docs/stable/torchvision/utils.html#torchvision.utils.make_grid>`_
+        scale_each: See `torchvision.utils.make_grid <https://pytorch.org/docs/stable/torchvision/utils.html#torchvision.utils.make_grid>`_
+        pad_value: See `torchvision.utils.make_grid <https://pytorch.org/docs/stable/torchvision/utils.html#torchvision.utils.make_grid>`_
+    """
+    def __init__(self,
+                 key=torchbearer.INPUT,
+                 transform=None,
+                 num_images=16,
+                 nrow=8,
+                 padding=2,
+                 normalize=False,
+                 norm_range=None,
+                 scale_each=False,
+                 pad_value=0):
+        super(MakeGrid, self).__init__(transform=transform)
+        self.key = key
+        self.num_images = num_images
+        self.nrow = nrow
+        self.padding = padding
+        self.normalize = normalize
+        self.norm_range = norm_range
+        self.scale_each = scale_each
+        self.pad_value = pad_value
+
+    def on_cache(self, cache, state):
+        import torchvision.utils as utils
+        return utils.make_grid(
+            cache,
+            nrow=self.nrow,
+            padding=self.padding,
+            normalize=self.normalize,
+            range=self.norm_range,
+            scale_each=self.scale_each,
+            pad_value=self.pad_value
+        )
+
+
+class FromState(ImagingCallback):
+    """The :class:`FromState` callback is an :class:`ImagingCallback` which retrieves and image from state when called.
+    The number of times the function is called can be controlled with a provided decorator (once_per_epoch, only_if
+    etc.)
+
+    Args:
+        key (:class:`.StateKey`): The key in state containing image (tensor of size [c, w, h])
+        transform (callable, optional): A function/transform that  takes in a Tensor and returns a transformed version.
+            This will be applied to the image before it is sent to output.
+        decorator: A function which will be used to wrap the callback function. once_per_epoch by default
+    """
+    def __init__(self, key, transform=None, decorator=once_per_epoch):
+        super(FromState, self).__init__(transform=transform)
+        self.key = key
+        self.on_batch = decorator(self.on_batch)
+
+    def on_batch(self, state):
+        if self.key in state:
```

Comment:
This should probably be a try catch for consistency

Reply:
Agreed, done :)